src/main/kotlin/open/tresorier/api/Api.kt

```diff
@@ -321,15 +321,25 @@ fun main() {
     app.get("/operation/account") { ctx ->
         val user = getUserFromAuth(ctx)
         val account: Account = ServiceManager.accountService.getById(user, getQueryParam<String>(ctx, "account_id"))
-        val operations = ServiceManager.operationService.findByAccount(user, account)
+        val categoryId: String? = getOptionalQueryParam<String>(ctx, "category_id")
+        var category: Category? = null
+        if (categoryId != null) {
+            category = ServiceManager.categoryService.getById(user, categoryId)
+        }
+        val operations = ServiceManager.operationService.findByAccount(user, account, category)
         ctx.json(operations)
     }
 
     app.before("/operation/budget", SuperTokens.middleware())
     app.get("/operation/budget") { ctx ->
         val user = getUserFromAuth(ctx)
         val budget: Budget = ServiceManager.budgetService.getById(user, getQueryParam<String>(ctx, "budget_id"))
-        val operations = ServiceManager.operationService.findByBudget(user, budget)
+        val categoryId: String? = getOptionalQueryParam<String>(ctx, "category_id")
+        var category: Category? = null
+        if (categoryId != null) {
+            category = ServiceManager.categoryService.getById(user, categoryId)
```

Comment:
correct but kotlin can help you make it in one line instead of five, example here : (l.161 same file)

val endMonth: Month? = getOptionalQueryParam<Int>(ctx, "end_month")?.let { Month.createFromComparable(it)}